optuna/samplers/_nsga2/sampler.py

```diff
@@ -154,18 +162,36 @@ def sample_relative(
         generation = parent_generation + 1
         study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
 
+        dominates = _dominates if self._constraints_func is None else _constrained_dominates
         if parent_generation >= 0:
-            p0 = self._select_parent(study, parent_population)
+            # crossover
             if self._rng.rand() < self._crossover_prob:
-                p1 = self._select_parent(
-                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                child = crossover(
+                    self._crossover_name,
+                    study,
+                    parent_population,
+                    search_space,
+                    self._rng,
+                    self._swapping_prob,
+                    dominates,
                 )
             else:
-                p1 = p0
+                child = parent_population[0].params
 
-            study._storage.set_trial_system_attr(
-                trial_id, _PARENTS_KEY, [p0._trial_id, p1._trial_id]
-            )
+            # mutation
+            params_len = len(child)
+            if self._mutation_prob is None:
+                mutation_prob = 1.0 / max(1.0, params_len)
+            else:
+                mutation_prob = self._mutation_prob
+
+            for param_name in child.keys():
+                param, param_distribution = child[param_name], search_space[param_name]
+                if param is None or self._rng.rand() < mutation_prob:
+                    child[param_name] = self._random_sampler.sample_independent(
+                        study, trial, param_name, param_distribution
+                    )
+            return child
```

Comment:
`search_space.keys()` is a subset of `child.keys()` since we calculate the search space by `intersection_search_space`. The element of `child.keys()` might be missing in `search_space.keys()` so`search_space[param_name]` might occur the `KeyError`.

How about the following?
```suggestion
            params = {}
            for param_name in child.keys():
                if self._rng.rand() >= mutation_prob:
                    # With probability 1 - mutation_prob, we return the parameter value of the child.
                    params[param_name] = child[param_name]
            return params
```

Reply:
I understand the cause of the `Key Error`.
With this fix, I think the individuals that are returned will have missing parameters due to the probability of `mutation_prob`.
In this case, is it correct that `sample_independent` will be called only for the missing parameter?

Reply:
Yes, I think so. The missing parameter will be sampled by `sample_independent`, which is fallen back to the `_random_sampler.sample_independent`. This is a same logic as before.